Devscord.DiscordFramework/Middlewares/Contexts/UserRole.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Devscord.DiscordFramework.Middlewares.Contexts
+{
+    public class UserRole
+    {
+        public ulong Id { get; private set; }
+        public string Name { get; private set; }
+        public Permissions Permissions { get; private set; }
+
+        public UserRole(string name)
+        {
+            this.Name = name;
+            this.Permissions = new Permissions();
+        }
+
+        public UserRole(string name, IEnumerable<Permission> permissions)
+        {
+            this.Name = name;
+            this.Permissions = new Permissions(permissions);
+        }
+
+        public UserRole(ulong id, string name, IEnumerable<Permission> permissions)
+        {
+            this.Id = id;
+            this.Name = name;
+            this.Permissions = new Permissions(permissions);
+        }
+    }
+
+    public class Permissions
+    {
+        private readonly IEnumerable<Permission> _permissions;
+        public ulong RawValue => (ulong)_permissions.Sum(x => (long)x);
+
+        public Permissions()
+        {
+            this._permissions = new List<Permission>();
+        }
+
+        public Permissions(IEnumerable<Permission> permissions)
+        {
+            this._permissions = permissions;
+        }
+
+        public List<Permission> ToList()
+        {
+            return _permissions.ToList();
+        }
+    }
+
+    public class ChangedPermissions
+    {
+        public Permissions AllowPermissions { get; }
+        public Permissions DenyPermissions { get; }
+
+        public ChangedPermissions(IEnumerable<Permission> allowPermissions, IEnumerable<Permission> denyPermissions)
+        {
+            AllowPermissions = new Permissions(allowPermissions);
+            DenyPermissions = new Permissions(denyPermissions);
+        }
+    }
+
+    public enum Permission
```

Comment:
=> osobny plik

Reply:
wrzuciłbym to do namespace z commons

np framework commons
albo watchman commons 

myśle że watchman commons będzie ok, bo to wspólne wartości dla discorda, które są niezależne od biblioteki/frameworka